libnetdata/popen/popen.h

```diff
@@ -8,6 +8,23 @@
 #define PIPE_READ 0
 #define PIPE_WRITE 1
 
+/* custom_popene_internal_dont_use_directly flag definitions */
+#define POPEN_FLAG_NONE        0
+#define POPEN_FLAG_CREATE_PIPE 1 // Create a pipe like popen() when set, otherwise set stdout to /dev/null
+#define POPEN_FLAG_CLOSE_FD    2 // Close all file descriptors other than STDIN_FILENO, STDOUT_FILENO, STDERR_FILENO
+
+// the flags to be used by default
+#define POPEN_FLAGS_DEFAULT (POPEN_FLAG_CREATE_PIPE|POPEN_FLAG_CLOSE_FD)
+
+// mypopen_raw is the interface to use instead of custom_popene_internal_dont_use_directly()
+// mypopen_raw will add the terminating NULL at the arguments list
+// we append the parameter 'command' twice - this is because the underlying call needs the command to execute and the argv[0] to pass to it
+#define mypopen_raw_default_flags_and_environment(pidptr, fpp, command, args...) custom_popene_internal_dont_use_directly(pidptr, environ, POPEN_FLAGS_DEFAULT, fpp, command, command, ##args, NULL)
+#define mypopen_raw_default_flags(pidptr, env, fpp, command, args...) custom_popene_internal_dont_use_directly(pidptr, env, POPEN_FLAGS_DEFAULT, fpp, command, command, ##args, NULL)
+#define mypopen_raw(pidptr, env, flags, fpp, command, args...) custom_popene_internal_dont_use_directly(pidptr, env, flags, fpp, command, command, ##args, NULL)
```

Comment:
One advise is to keep the line width with in 80 chars limit whenever possible.

Reply:
I personally find 80 way too narrow. Probably 120 is better.
But anyway, this is just cosmetic.